### DISCUSSION TITLE
Bug? w-full doesn't work on mobile safari

### DISCUSSION THREAD

**Discussion by tschuehly** [tailwindlabs/tailwindcss#10561] [Help] [answered]
Hello everyone

I have an issue where I have an input field that is inside a label that is flex.

```
<label class="flex w-full cursor-pointer flex-col px-4 sm:flex-row">
  <span class="mb-2 w-full pl-4 pr-4 font-semibold sm:w-80 sm:pl-0">Datum der Hochzeit</span>
  <input class="w-full grow" pattern="[0-9]{1,20}" required="required" x-ref="date" type="date" id="date" name="date" value="" />
</label>
```

This works on chrome but doesn't work on mobile safari. I made a tailwind play to test it: https://play.tailwindcss.com/0IgiWfIsKi

Chrome: 
![image](https://user-images.githubusercontent.com/33346637/218318196-9b5ea412-cb1f-4be2-bf4e-aeeb542151d3.png)

Mobile Safari:
![image](https://user-images.githubusercontent.com/33346637/218318418-1f76e45b-7777-4e85-b23b-0321a58d242e.png)

I can fix the issue by overwriting w-full:

```
.w-full{
    width: 100%;
    min-width: intrinsic;
}
```

![image](https://user-images.githubusercontent.com/33346637/218318512-5dcb3497-207d-4fe7-bac8-aa026e622fac.png)

Is this intended Behaviour?

**Comment by adamwathan** [comment, marked as answer]
Hey! If this behaves differently on mobile Safari it's just one of those weird CSS quirks you have to learn about and deal with just like you would if you were writing vanilla CSS. The `w-full` class just sets `width: 100%`, so whatever happens in the browser is exactly what would happen if you added that CSS property to that element yourself and that's intentional. So the solution here is just do whatever you'd have to do to get the results you want with CSS but using utility classes. Hope that makes sense!

**Reply by tschuehly** [reply]
Ok thank you for the info
